### PR TITLE
fit_1d_gaussian_and_solve_coeff

### PR DESCRIPTION
**Pull request by Winnie08L**

### CODE REVIEW

File: surface_potential_analysis/kernel/kernel.py [hunk 1]
```diff
@@ -658,3 +667,52 @@ def truncate_noise_kernel(
...
+            )
+            for m in np.arange(0, n + 1)
+        ],
+    )
+    coeff = np.linalg.solve(coeff_to_solve, true_noise_coeff)
+    coeff = coeff.T
+    return np.concatenate([coeff, coeff[1:]])
+
+
+def get_lorentzian_parameter(
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/kernel.py:694]: I think a 'get effective Lorentzian parameter' (similar to gaussian which matches the Taylor expansion of the noise to caldera Leggett) would be more helpful here

File: surface_potential_analysis/kernel/kernel.py [hunk 2]
```diff
@@ -658,3 +667,52 @@ def truncate_noise_kernel(
...
+            np.array(
+                [
+                    (((-1) ** m) / (factorial(2 * m))) * ((i * delta_k) ** (2 * m))
+                    for i in np.arange(0, n + 1)
+                ],
+            )
+            for m in np.arange(0, n + 1)
+        ],
+    )
+    coeff = np.linalg.solve(coeff_to_solve, true_noise_coeff)
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/kernel.py:689]: same comments about naming coeff

**Review by Matt-Ord — CHANGES_REQUESTED** [review]
great start - left a few comments. This is really great work and hopefully unlocks 2d simulations for us!

**Comment by Matt-Ord** [comment]
@Winnie08L left a review - but you will have to fix conflicts first. I would suggest rebasing on top of the commits from my branch

File: surface_potential_analysis/kernel/kernel.py [hunk 3]
```diff
@@ -658,3 +667,52 @@ def truncate_noise_kernel(
             "eigenvalue": operators["eigenvalue"][args],
         }
     )
+
+
+def get_coefficient_matrix_taylor(
+    *,
+    true_noise_coeff: np.ndarray[tuple[int, int], np.dtype[np.float64]],
+    delta_k: np.floating[Any],
```
[the remark this answers is not in the document]
Comment (Winnie08L) [reply surface_potential_analysis/kernel/kernel.py:675]: but output of np.linalg.norm is floating and this is used to find delta_k, and floating is incompatible with foalt

**Pull request closed without merging**